Microsoft.Identity.Web/WebApiServiceCollectionExtensions.cs

```diff
@@ -41,13 +41,9 @@ public static class WebApiServiceCollectionExtensions
             string configSectionName = "AzureAD",
             bool subscribeToJwtBearerMiddlewareDiagnosticsEvents = false)
         {
-            services.Configure<AzureADOptions>(options => configuration.Bind(configSectionName, options));
             services.AddAuthentication(AzureADDefaults.JwtBearerAuthenticationScheme)
                     .AddAzureADBearer(options => configuration.Bind(configSectionName, options));
 
-            // Add session if you are planning to use session based token cache , .AddSessionTokenCaches()
```

Comment:
I agree.
@kalyankrishna1 : does it make sense to have a session in Web apis?

Reply:
Kalyan suggested me to remove this line (and it is even commented).

Reply:
Because I'm trying to minimize Azure hosting costs, I wish to have all necessary state stored in cookies rather than back the session/token store with a database or distributed cache.  Does this change basically require such a persistence?  If all I need to do is know the user's verified identity, is cookie storage impractical?

Following the 1-WebApp-OIDC sample verbatim, I'm able to log in and navigate for a while, but after some time - a few minutes or a few hours, not sure yet, but definitely not days.

Is it upon AppService idle?  Manual AppService restart (probably, but the timing is not perfectly aligned)?  Session Timeout?  Are session keys being reset and no longer valid when Azure re-balances where the app service is hosted?  I'm having trouble diagnosing it.  Trying `services.Configure<SessionOptions>(options => { options.IdleTimeout = TimeSpan.FromMinutes(1); });` seems to have no effect - I'm not logged out after 1 minute of zero activity.

At such times of being unexpectedly logged out, `User.Identity.IsAuthenticated` is false.  My application attempts an interactive login, but AzureAD redirects back immediately and logs me in.  This would be less noticeable/disruptive if I weren't in an iframe requiring a popup window for the login sequence.  (I'm pretty sure the popup behavior has nothing to do with being forgotten early.)

I'm sure I'm missing some important concepts, but there seems to be a lot of fact collisions between non-core and core and between versions and among alternatives to this AzureSamples/Microsoft.Identity.Web approach.  It's hard to know what documentation/SO posts are relevant for this scenario.

Reply:
@JasonKleban : if you are building a Web API (not a web app), the sample to have a look to is https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2.